### PR TITLE
fix: get git diff all at once

### PR DESCRIPTION
**Pull request by froucher**
Get all git differences all at once

Issue #263

### CODE REVIEW

File: lib/cli/changeset.js [hunk 1]
```diff
@@ -71,20 +72,27 @@ SubCommand.prototype.process = function(proc, callback) {
...
-		.pipe(Diff.stream({
-			ignoreWhitespace: !!self.opts['--ignore-whitespace']
-		}))
-		.pipe(MetadataContainer.diffStream())
+	var diff = {}; // first get the stdin stream with all git diff changes
+	proc.stdin.pipe(miss.concat(function(result) {
+		diff = Diff.parse(result, { ignoreWhitespace: !!self.opts['--ignore-whitespace'] });
+	}));
 
-	mergeStream(stdin, metadataContainer.getStream())
```
Comment (froucher) [inline lib/cli/changeset.js:80]: @amtrack, why `mergeStream` exists? I removed this merge and it seems to be working...
If it is not needed, then we can add a commit removing `metadataContainer.getStream()` function.

**Comment by froucher** [comment]
Hello @amtrack,

How are you? Maybe it could be helpful to have a session to review this PR. Please let me know if it is interesting from you side.

Regards and thank you

**Comment by amtrack** [comment]
> Hello @amtrack,
> 
> How are you? Maybe it could be helpful to have a session to review this PR. Please let me know if it is interesting from you side.
> 
> Regards and thank you

@froucher Thanks, I'm doing well.
Yes, let's do a session. I'll contact you via email regarding an appointment.

**Comment by amtrack** [comment]
@froucher I have one minor change request, besides that this PR looks good to me.

Apparently the `metadataContainer` constructed [in these lines](https://github.com/amtrack/force-dev-tool/blob/8d4fbc6a14a512511a91a651d669111bed81de7f/lib/cli/changeset.js#L53-L72) isn't used anymore by your implementation, so that explicitly creating changesets using command line arguments (e.g. `force-dev-tool changeset create mytestclass ApexClass/FooTest`) doesn't work anymore.
I've now relocated the corresponding test from `test-integration` to `test` (#267) so that it is run in CI for Pull Requests.

Can you please rebase your PR to latest master branch and try to integrate the `metadataContainer` instance again?

**Comment by froucher** [comment]
Hi,

Done, it fails...

A previous question, when I wanted to create a changeset with one file, I usually executed a command like:

```
git diff master..HEAD -- ApexClass/FooTest | force-dev-tool changeset create mytestclass
```

Maybe this workaround could replace the functionality.

Anyway, for backward compatibility, I will review the source code to understand why is not supported.

Thanks.

**Comment by froucher** [comment]
Hi @amtrack,

I resolved the issue about explicitly creating changeset using command line arguments.

Sorry for the delay.

**Comment by froucher** [comment]
Thank you! 👏 👏 👏

**Pull request merged**